### PR TITLE
Add maxWidth/maxHeight/maxArea, clarify compliance

### LINKED ISSUE #620: Limiting max height and width that can be requested
See the [thread started by Glen yesterday](https://groups.google.com/forum/#!topic/iiif-discuss/j-tg5F7J6VM). We've heard it before, and I was about to add my own slightly different use case, but @tomcrane has already done so. I'm perfectly happy to give you `full` of anything, but from a practical, systems perspective, I can't do it because the full image is just too large (i.e. scrolls, and some other huge prints we have) and the server will time out.

Yes, practically speaking, if you can get the tiles you can recreate the whole image. However, we've heard enough times now that it would make people comfortable and further adoption if we could find a solution.

The cheap solution would be to add a predicate like `fullAvailable: true|false`, and force clients to work with `sizes` to determine what they can get. The better solution, in my opinion, is something like the discussion around #277.

### PR DESCRIPTION
**Pull request by zimeon**
Add maxWidth/maxHeight/maxArea, clarify compliance document regarding support for combinations of parameters, number compliance document sections following form of Image API, give Level 0 its own section. Fixes #620

### CODE REVIEW

**Comment by zimeon** [comment]
Not sure what is up with the auto-website: http://maxwidth.iiif.io/api/image/2.1/ and http://maxwidth.iiif.io/api/image/2.1/compliance.html do not show the changes in this branch

**Comment by zimeon** [comment]
Auto-website now happy, see http://maxwidth.iiif.io/api/image/2.1/ and http://maxwidth.iiif.io/api/image/2.1/compliance.html

**Comment by azaroth42** [comment]
Can we add maxWidth (at least, as most common use case) to the example?

**Comment by zimeon** [comment]
Good point @azaroth42, added `maxWidth` in profile example

**Comment by azaroth42** [comment]
I think it's clear that if you specify both area and maxW/H that both apply, but is it worth making it explicit? Either way :+1: from me

File: source/api/image/2.1/compliance.md [hunk 1]
```diff
@@ -46,21 +48,25 @@ In the tables below "![required][icon-req]" indicates that support is _REQUIRED_
...
+| ,h     | `sizeByH`        | ![optional][icon-opt] | ![required][icon-req] | ![required][icon-req] |
+| pct:x  | `sizeByPct`      | ![optional][icon-opt] | ![required][icon-req] | ![required][icon-req] |
+| w,h    | `sizeByForcedWh` | ![optional][icon-opt] | ![optional][icon-opt] | ![required][icon-req] |
+| !w,h   | `sizeByWh`       | ![optional][icon-opt] | ![optional][icon-opt] | ![required][icon-req] |
+|        | `sizeAboveFull`  | ![optional][icon-opt] | ![optional][icon-opt] | ![optional][icon-opt] |
 {: .api-table}
 
-### Rotation
+Note that servers may express limits on the sizes available for an image with the optional `maxWidth`, `maxHeight` and/or `maxArea` [Profile Description properties][profile]. Servers are compliant provided they support the forms of the Size parameter shown above for image sizes up to the limits specified. Clients _SHOULD NOT_ assume that Region and Size parameter combinations such as `/full/full/` will be supported. 
+
```
Comment (jpstroop) [inline source/api/image/2.1/compliance.md]: Small thing, but why SHOULD NOT as opposed to MUST NOT?

Comment (jpstroop) [inline source/api/image/2.1/compliance.md]: And should we be telling clients what to do at all? Maybe we leave the warning text in place but remove the RFC _SHOULD NOT_?

Comment (zimeon) [inline source/api/image/2.1/compliance.md]: I agree, I think the correct thing is to make this a plain English "should not"

**Comment by jpstroop** [comment]
:+1:

**Comment by azaroth42** [comment]
:+1: (post deRFCification) There's no test or warning possible, so SHOULD NOT bears no real weight.

**Comment by zimeon** [comment]
@mikeapp @tomcrane - are we good on this one?

**Comment by tomcrane** [comment]
:+1:

**Comment by mikeapp** [comment]
:+1:

**Pull request merged**